### ISSUE TITLE
Using array index as key

### ISSUE THREAD

**Issue by SimonEvertree** [Balastrong/tanstack-form-demo#1]
https://github.com/Balastrong/tanstack-form-demo/blob/27728b541c20b448631772081faf73ac10eb088a/src/components/SignUp.tsx#L181

Tried to replicate what you've built but running into this issue: biome linter complaining that "Avoid using the index of an array as key property in an element." 
From what I can read it's ok to use the index as key _unless_ you plan on adding/removing items or changing the order. Both of which are happening in your example.

Do you have an idea for how to generate a new blank input, and not use the index as key? I cannot figure it out.

**Comment by Balastrong** [comment]
I agree there should be a better way to handle that, we also have a discussion here mentioning a unique id for each element: https://github.com/TanStack/form/discussions/1007

I'd say all contributions (ideas, discussions, pull requests) are welcome :)
